### PR TITLE
[SYCL] Refreshing event methods

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/source/detail/event_impl.cpp [hunk 1]
```diff
@@ -84,13 +84,23 @@ void event_impl::wait(
...
+#else
     simple_scheduler::Scheduler::getInstance().waitForEvent(Self);
+#endif
 }
 
 void event_impl::wait_and_throw(
     std::shared_ptr<cl::sycl::detail::event_impl> Self) {
   wait(Self);
+#ifdef SCHEDULER_20
+  for (std::shared_ptr<cl::sycl::detail::event_impl> &EventImpl :
```
Comment (keryell) [inline sycl/source/detail/event_impl.cpp]: Perhaps `auto` is enough, except if you really want to remind that this type is important to know.

File: sycl/source/event.cpp [hunk 2]
```diff
@@ -50,8 +50,18 @@ void event::wait_and_throw(const vector_class<event> &EventList) {
 }
 
 vector_class<event> event::get_wait_list() {
+#ifdef SCHEDULER_20
+  vector_class<event> Result;
+
+  for (std::shared_ptr<cl::sycl::detail::event_impl> &EventImpl :
```
Comment (keryell) [inline sycl/source/event.cpp]: Perhaps `auto` is enough, except if you really want to remind that this type is important to know.

Comment (romanovvlad) [reply sycl/source/event.cpp]: Done.

File: sycl/source/detail/event_impl.cpp (hunk 1 above)
Comment (romanovvlad) [reply sycl/source/detail/event_impl.cpp]: Done.

**Pull request merged**